test/src/unit-ordered-dim-label-reader.cc

```diff
@@ -878,3 +878,219 @@ TEST_CASE_METHOD(
+    write_labels(11, 30, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
+                          1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0});
+    write_labels(
+        15, 26, {0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.8});
+  }
+
+  SECTION("Tile aligned validate min") {
+    write_labels(16, 20, {0.1, 0.2, 0.3, 0.4, 0.5});
+    write_labels(21, 25, {0.4, 0.7, 0.8, 0.9, 1.0});
+  }
+
+  SECTION("Non tile aligned, contiguous, validate min") {
+    write_labels(16, 21, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6});
+    write_labels(22, 25, {0.5, 0.8, 0.9, 1.0});
+  }
+
+  SECTION("Tile aligned, overlapped, validate min") {
+    write_labels(6, 15, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0});
+    write_labels(11, 20, {0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3});
+  }
+
+  SECTION("Tile aligned validate max") {
+    write_labels(21, 25, {0.4, 0.7, 0.8, 0.9, 1.0});
+    write_labels(16, 20, {0.1, 0.2, 0.3, 0.4, 0.5});
+  }
+
+  SECTION("Non tile aligned, contiguous, validate max") {
+    write_labels(22, 25, {0.5, 0.8, 0.9, 1.0});
+    write_labels(16, 21, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6});
+  }
+
+  SECTION("Tile aligned, overlapped, validate max") {
+    write_labels(16, 25, {0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3});
+    write_labels(11, 20, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0});
+  }
+
+  REQUIRE_THROWS_WITH(
+      read_labels({0.8, 0.9}),
+      "Error: Internal TileDB uncaught exception; ReaderBase: Attribute out of "
+      "order");
+}
+
+TEST_CASE_METHOD(
+    CPPOrderedDimLabelReaderVarFx,
+    "Ordered dimension label reader: var labels, out of order",
+    "[ordered-dim-label-reader][var][out-of-order]") {
+  SECTION("Non tile aligned, overlapped 1") {
+    write_labels(11, 20, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0});
+    write_labels(15, 16, {0.3, 0.6});
+  }
+
+  SECTION("Non tile aligned, overlapped 2") {
+    write_labels(11, 20, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0});
+    write_labels(15, 16, {0.5, 0.8});
+  }
+
+  SECTION("Non tile aligned, overlapped 3") {
+    write_labels(11, 30, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
+                          1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0});
+    write_labels(
+        15, 26, {0.3, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6});
+  }
+
+  SECTION("Non tile aligned, overlapped 4") {
+    write_labels(11, 30, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
+                          1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0});
+    write_labels(
+        15, 26, {0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.8});
+  }
+
+  SECTION("Tile aligned validate min") {
+    write_labels(16, 20, {0.1, 0.2, 0.3, 0.4, 0.5});
+    write_labels(21, 25, {0.4, 0.7, 0.8, 0.9, 1.0});
+  }
+
+  SECTION("Tile aligned validate min, equality") {
+    write_labels(16, 20, {0.1, 0.2, 0.3, 0.4, 0.5});
+    write_labels(21, 25, {0.5, 0.7, 0.8, 0.9, 1.0});
+  }
```

Comment:
I threw one test in that checks an error is raised when the ordering has equality to make sure a fix I added was correct, but we probably want to add a couple more of these.